### DISCUSSION TITLE
Send setremote commands from python script

### DISCUSSION THREAD

**Discussion by dtlr7** [jgyates/genmon#1098] [Questions] [answered]
### Checklist

- [X] I made sure that there are *no existing issues or discussions* - [open issues](https://github.com/jgyates/genmon/issues),[closed issues](https://github.com/jgyates/genmon/issues?q=is%3Aissue+is%3Aclosed),[open discussion](https://github.com/jgyates/genmon/discussions), [closed discussion](https://github.com/jgyates/genmon/discussions/categories/questions?discussions_q=is%3Aclosed) - which may answer my question or that I could contribute my information to.
- [X] I have read the [Known Issues Wiki Page](https://github.com/jgyates/genmon/wiki/Appendix-D-Known-Issues) and my question / issue is not listed.
- [X] I have provided the version of the software and the operating system in my post.
- [X] If I have an issue with genmon not communicating with the controller, I have read and followed the [serial troubleshooting guide](https://github.com/jgyates/genmon/wiki/3.6---Serial-Troubleshooting).

### Version

V1.19.02

### Have you submitted your log files from the About page? Note that outbound email must be working for logs to be submitted.

I have not submitted logs

### What is/are your question(s)?

Any ideas on how to send a setremote command via a python script running on the same raspberry pi?

I've tried using the web api but get an unknown command response.

**Comment by jgyates** [comment, marked as answer]
The [ClientInterface.py](https://github.com/jgyates/genmon/blob/master/ClientInterface.py) program in the root folder of the project will take what the user types and send it to genmon as a command. This uses the library file in [./genmon/genmonlib/myclient.py](https://github.com/jgyates/genmon/blob/master/genmonlib/myclient.py) to handle the TCP socket interface for sending the command.

To use the web interface you can use [this example](https://github.com/jgyates/genmon/wiki/Appendix-C--Interfacing-Generator-Monitor-to-External-Applications#web-api). This example assumes you are using HTTPS with a user name and password. The URL used for the setremote command would be something like this:

http://localhost:8000/cmd/setremote?setremote=off

change "off" to the remote command you want to use. The main options are for the commands are start,stop and starttransfer. Other commands are in the code here:

https://github.com/jgyates/genmon/blob/f849906c76564c7dc809f1d7cd7dc1c5c7bf1880/genmonlib/generac_evolution.py#L1529

however not all of them work on all firmware and controller variants.

Let me know if you have any questions.

**Comment by jgyates** [comment]
one clarifiaction, the "off" command is only for Evolution Liquid Cooled models. start, stop and starttransfer are the main commands to start, stop and start with the transfer switch activated.
